transient ut failure NamespaceExtractionCacheManagerExecutorsTest

Build: https://travis-ci.org/druid-io/druid/jobs/74628709
```
Running io.druid.server.namespace.cache.NamespaceExtractionCacheManagerExecutorsTest

Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.494 sec <<< FAILURE! - in io.druid.server.namespace.cache.NamespaceExtractionCacheManagerExecutorsTest

testDelete(io.druid.server.namespace.cache.NamespaceExtractionCacheManagerExecutorsTest)  Time elapsed: 0.049 sec  <<< FAILURE!

java.lang.AssertionError: null

	at org.junit.Assert.fail(Assert.java:86)

	at org.junit.Assert.assertTrue(Assert.java:41)

	at org.junit.Assert.assertTrue(Assert.java:52)

	at io.druid.server.namespace.cache.NamespaceExtractionCacheManagerExecutorsTest.testDelete(NamespaceExtractionCacheManagerExecutorsTest.java:316)

	at io.druid.server.namespace.cache.NamespaceExtractionCacheManagerExecutorsTest.testDelete(NamespaceExtractionCacheManagerExecutorsTest.java:258)
```

--- a/extensions/namespace-lookup/src/test/java/io/druid/server/namespace/cache/NamespaceExtractionCacheManagerExecutorsTest.java
+++ b/extensions/namespace-lookup/src/test/java/io/druid/server/namespace/cache/NamespaceExtractionCacheManagerExecutorsTest.java
@@ -280,6 +280,7 @@ public class NamespaceExtractionCacheManagerExecutorsTest
         null
     );
     final String cacheId = UUID.randomUUID().toString();
+    final CountDownLatch latchBeforeMore = new CountDownLatch(1);
     ListenableFuture<?> future =
         manager.schedule(
             namespace, factory, new Runnable()
@@ -301,17 +302,29 @@ public class NamespaceExtractionCacheManagerExecutorsTest
                 }
                 finally {
                   latch.countDown();
-                  latchMore.countDown();
+                  try {
+                    if (latch.getCount() == 0) {
+                      latchBeforeMore.await();
+                    }
+                  }
+                  catch (InterruptedException e) {
+                    log.debug("Interrupted");
+                    Thread.currentThread().interrupt();
+                  }
+                  finally {
+                    latchMore.countDown();
+                  }
                 }
               }
             },
             cacheId
         );
     latch.await();
+    prior = runs.get();
+    latchBeforeMore.countDown();
     Assert.assertFalse(future.isCancelled());
     Assert.assertFalse(future.isDone());
     Assert.assertTrue(fnCache.containsKey(ns));
-    prior = runs.get();
     latchMore.await();
     Assert.assertTrue(runs.get() > prior);
 

